ros2param/test/test_cli.py

```diff
@@ -0,0 +1,718 @@
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_list_unexisting_node(self):
+        with self.launch_node_command(
+                arguments=['list', '/unexisting_node']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Node not found'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_list_hidden_existing_node_no_hidden_flag(self):
+        with self.launch_node_command(
+                arguments=['list', '/_hidden_param_node']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == 1
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Node not found'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_list_hidden_existing_node_hidden_flag(self):
+        with self.launch_node_command(
+                arguments=[
+                    'list', '/_hidden_param_node', '--include-hidden-nodes']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_text=PARAM_NODE_PARAMETER_LIST,
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_list_existing_node_bool_prefix(self):
+        with self.launch_node_command(
+                arguments=[
+                    'list', '--param-prefixes=bool', '/param_node']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_text=PARAM_NODE_BOOL_PARAMETER_LIST,
+            text=node_command.output,
+            strict=True
+        )
+
+    # TODO(BMarchi): This test fails for opensplice only. It hangs
+    # in wait_for_shutdown method. It seems that it doesn't return
+    # after getting all the nodes with their parameters.
+    # @launch_testing.markers.retry_on_failure(times=3)
+    # def test_list_all_nodes_all_parameters(self):
+    #    with self.launch_node_command(
+    #            arguments=['list', '--include-hidden-nodes']) as node_command:
+    #        assert node_command.wait_for_shutdown(timeout=5)
+    #    assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+    #    assert launch_testing.tools.expect_output(
+    #        expected_text=ALL_NODES_PARAMETER_LIST,
+    #        text=node_command.output,
+    #        strict=True
+    #    )
```

Comment:
@BMarchi if it only happens with Opensplice, consider checking `rmw_implementation` value and using [`skipTest` ](https://docs.python.org/3.6/library/unittest.html#unittest.TestCase.skipTest) appropriately instead of commenting out the entire test.

Reply:
Turns out it happens in every implementation, I can't remember if it worked before.

Reply:
No, only launching the process

Reply:
Hmm, interesting.